src/utils/helper_functions.R

```diff
@@ -0,0 +1,79 @@
+# NOTE: These helper functions are ports of the original Python functions in 'helper_functions.py'
+
+#' Adds annotations (.var and .obs) from the unintegrated dataset to the
+#' integrated dataset. In the case of the control method "perfect_integration",
+#' the function will fetch annotations from the validation dataset instead.
+#'
+#' @param i_adata AnnData object, batch-integrated dataset
+#' @param v_adata AnnData object, validation dataset
+#' @param u_adata AnnData object, unintegrated dataset
+#' @return AnnData object with .var and .obs added
+get_obs_var_for_integrated <- function(i_adata, v_adata, u_adata) {
+  if (i_adata$uns$method_id == "perfect_integration") {
```

Comment:
we have 2 perfect integration control methods now, vertical and horizontal. The stuff in this if statement is only applicable to perfect_integration_horizontal. Change the "perfect_integration" to "perfect_integration_horizontal". 

Will have to create a separate if statement for the perfect_integration_vertical. I'll make an issue once this PR is merged.